### ISSUE TITLE
Pulled RTMP Endpoint resolution is constant in 480p

### ISSUE THREAD

**Issue by SelimEmre** [ant-media/Ant-Media-Server#2332]
### Short description
Here is scenario: Add RTMP Endpoint in WebRTC Stream. Check added RTMP Endpoint resolution value with ffprobe. You can see the resolution is 640x480. I tried different cases and I encountered this issue always.

### Steps to reproduce
1. Add 2 stream which is testStream1 and testStream2
2. Add testStream2 RTMP Endpoint in testStream1.
3. Publish WebRTC Stream in testStream1
4. Check testStream2 resolution parameter with ffprobe.
5. You can see 640x480 parameters

### Expected behavior
If we send 1080p with WebRTC, RTMP Endpoint should have a 1080p resolution.

### Actual behavior
RTMP Endpoint resolution 480p is fixed at whatever resolution I send it with WebRTC.

### Logs
AMS 2.1 SNAPSHOT
I'm using default settings.

FFprobe output:
```
Input #0, flv, from 'rtmp://domain.com/WebRTCAppEE/4OaT5dGcK2Vu1594936913510':
  Metadata:
    encoder         : Lavf58.20.100
    length          : 12
  Duration: 00:00:00.00, start: 0.000000, bitrate: N/A
    Stream #0:0: Video: h264 (Constrained Baseline), yuv420p(progressive), 640x480, 1k tbr, 1k tbn
    Stream #0:1: Audio: aac (LC), 44100 Hz, stereo, fltp, 96 kb/s
```

**Comment by mekya** [comment]
Thank you for your question. 

WebRTC changes the resolution according to the bandwidth. Opening camera with 1080 does not mean that WebRTC send stream with 1080p. It adapts the resolution according to the bandwidth and it can even send lower than 480p.

Btw, according to the webrtc standards, there are some more settings coming that gives more options to developer to maintain resolution or frame rate.

**Comment by kcs-girishsolanki** [comment]
It adapts the resolution according to the bandwidth and it can even send lower than 480p. - I have 20 MBPS internet speed even after antmedia sending 480P to you tube.

**Comment by mekya** [comment]
Thank you for your comment. 

TL;DR;

- I can recommend increasing the max bandwidth in WebRTC Adaptor. It's 900Kbps by default. [You can change to upper value.](https://github.com/ant-media/StreamApp/blob/b4b7a4a93caf84424590bb4b2fbfcc755debca64/src/main/webapp/index.html#L327)
- If there is no adaptive streaming, Ant Media Server sends the incoming stream to the endpoint. It means that it just relays the same stream without changing its quality and resolution.  

Let me point out some more things for enthusiasts 

- The real bandwidth is between server and client.  20Mbps internet speed is the speed between you and your service provider. After your request pass the service provider, your request is in the Internet and your internet provider cannot guarantee internet speed to everywhere with 20Mbps. Because there may be even some bottleneck in the route or they may be even some bottleneck in the data center in which your server resides. 

- WebRTC stack has the hard coded upper limit 2500kbps for streams. I mean even if you gigabit network, browser does not use more bandwidth than the 2500kbps for streams. Because, from my point of view, the Internet does not have QoS for Real Time Communication. If you have average gigabit network speed for one hour, you can see some kind of jitter, pixelation during the streaming. Because average network speed does not guarantee that you have the same speed for all the time without any jitter or packet lost.  In RTC, even sub-second network fluctuations may affect the quality.

**Comment by kcs-girishsolanki** [comment]
Okay.. I got it but the thing is when i start the streaming through out whole process it look like same there is no change in resolution even i am setting resolution to 1080 it's always taking lower level resolution.

Can you please help into that I having big issue in my live application where thousand of users are there.

**Comment by mekya** [comment]
Sorry for late reply. 

It's something about browser side. It decides which resolution to send.  The resolution may increase over the time but the browser decides that  

You can google about that for having fixed and high quality in browser for WebRTC streaming. Maybe I'm or we're missing somethings

**Comment by jantekb** [comment]
Closed due to no activity
